synapse/api/errors.py

```diff
@@ -510,7 +511,12 @@ def __init__(
         retry_after_ms: Optional[int] = None,
         errcode: str = Codes.LIMIT_EXCEEDED,
     ):
-        super().__init__(code, msg, errcode)
+        headers = (
+            None
+            if retry_after_ms is None
+            else {"Retry-After": str(math.ceil(retry_after_ms / 1000))}
+        )
+        super().__init__(code, msg, errcode, None, headers)
```

Comment:
Please use keyword args for the last two (I'm left wondering what the `None` is for).